src/middlewared/middlewared/plugins/system.py

```diff
@@ -1442,39 +1527,36 @@ def set_crash_reporting(self):
 
 
 async def _update_birthday_data(middleware, birthday=None):
-    middleware.logger.debug('Synchronization/update birthday data')
-    # Check if it is exists already
-    system_obj = await middleware.call('system.info')
-    birthday_obj = system_obj['birthday']
-    if birthday_obj is not None:
-        # Already setted before.
+
+    birthday = (await middleware.call('system.info'))['birthday']
+    if birthday is not None:
+        # already been set
         return
 
-    # If it is not defined yet, it will try to define
+    # get current time and use as birthday
     if birthday is None:
         birthday = await middleware.call('system.sync_clock')
 
-    if birthday is not None:
-        # Update System Settings
+        middleware.logger.debug('Updating birthday data')
+        # update db with new birthday
         settings = await middleware.call('datastore.config', 'system.settings')
         await middleware.call('datastore.update', 'system.settings', settings['id'], {
             'stg_birthday': birthday,
         })
 
 
-# Update Birthday Date
 async def _update_birthday(middleware):
-    # Sync clock
-    middleware.logger.debug('Synchronization the clock for system birthday')
+
     birthday = None
     timeout = 3600 * 24
 
     middleware.register_hook('interface.post_sync', _update_birthday_data)
 
+    middleware.logger.debug('Waiting for clock sync to update system birthday')
     while birthday is None:
         birthday = await middleware.call('system.sync_clock')
 
-        # Wait until be able to sync the clock
+        # sleep for 1 day and try again
```

Comment:
We should not wait for 1 day, i think a delay of 15 min or 30 min should be enough to let ntp sync clock